<launch>
	<!-- MAVROS Parameters -->
    <arg name="fcu_url" default="udp://127.0.0.1:14540@:14550" />
    <arg name="gcs_url" default="" />
    <arg name="tgt_system" default="1" />
    <arg name="tgt_component" default="1" />
	<arg name="fcu_protocol" default="v2.0" />
	<arg name="respawn_mavros" default="false" />
	<arg name="namespace" default="mavros"/>

	<node pkg="mavros" exec="mavros_node" namespace="$(var namespace)" output="screen" >
		<param name="fcu_url" value="$(var fcu_url)" />
		<param name="gcs_url" value="$(var gcs_url)" />
		<param name="tgt_system" value="$(var tgt_system)" />
		<param name="tgt_component" value="$(var tgt_component)" />
		<param name="fcu_protocol" value="$(var fcu_protocol)" />
        <param name="respawn_mavros" value="$(var respawn_mavros)" />
	</node>

	<!-- Flight Patterns
	0: CIRCULAR
	1: SPIRAL
	2: CLOUD (frequency 7.0 for clear pattern)
	3: SINE (speed 0.3, frequency 3.0 for displacement of 1.26)
	4: N-GRAM (hepta-gram, vertices 7)
	-->
	<arg name="flight_pattern" default="2"/>
	<arg name="max_iter" default="2"/>
	<!-- Flight Parameters -->
	<arg name="dt" default="0.05"/>
	<arg name="radius" default="0.80"/>
	<arg name="height" default="1.00"/>
	<arg name="speed" default="0.30"/>
	<arg name="min_speed" default="0.05"/>
	<arg name="offset_x" default="0.00"/>
	<arg name="offset_y" default="0.00"/>
	<arg name="offset_z" default="0.50"/>
	<arg name="frequency" default="7.00"/>
	<arg name="ngram_vertices" default="7"/>
	<arg name="ngram_step" default="2"/>

    <node pkg="mavros_offb" exec="mavros_offb_node" name="mavros_offboard" output="screen">
    	<param name="flight_pattern" value="$(var flight_pattern)" />
		<param name="max_iter" value="$(var max_iter)" />
		<param name="height" value="$(var height)" />
		<param name="dt" value="$(var dt)" />
		<param name="radius" value="$(var radius)" />
		<param name="speed" value="$(var speed)" />
		<param name="min_speed" value="$(var min_speed)" />
		<param name="offset_x" value="$(var offset_x)" />
		<param name="offset_y" value="$(var offset_y)" />
		<param name="offset_z" value="$(var offset_z)" />
		<param name="frequency" value="$(var frequency)" />
		<param name="ngram_vertices" value="$(var ngram_vertices)" />
		<param name="ngram_step" value="$(var ngram_step)" />
	</node>  

</launch>
